components/src/core/components/Input/Select/SelectInput.vue

```diff
@@ -182,10 +186,37 @@ export default defineComponent({
     },
   },
 
+  methods: {
+    scrollToOptionByIndex(index: number) {
+      this.$nextTick(() => {
+        let option = this.$refs[`option-${index}`] as any;
+        // Handle array refs in v-for
+        if (Array.isArray(option)) {
+          option = option[0];
+        }
+        const el = option?.$el || option;
+        if (el && el.scrollIntoView) {
+          this.scrollToView(el);
+        }
+      });
+    },
+  },
+
   watch: {
     pointer(newIndex: number) {
-      const option = this.$refs[`option-${newIndex}`];
-      if (option?.$el) this.scrollToView(option.$el);
+      if (newIndex >= 0 && this.dropdownOpen) {
+        this.$nextTick(() => {
+          let option = this.$refs[`option-${newIndex}`] as any;
+          // Handle array refs in v-for
+          if (Array.isArray(option)) {
```

Comment:
Not sure why this check is needed? please debug and check if this is something truly needed, if not remove